lib/puppet/type/group.rb

```diff
@@ -108,7 +110,9 @@ def is_to_s(currentvalue)
     end
 
     newparam(:auth_membership, :boolean => true, :parent => Puppet::Parameter::Boolean) do
-      desc "whether the provider is authoritative for group membership."
+      desc "Whether the provider is authoritative for group membership. This
+        must be set to true to allow setting the group to empty e.g.
+        `members => [],`."
```

Comment:
"empty **value**" doesn't really make sense in the context of Groups. It's a bucket, not an object.